geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/misc/WANSSLDUnitTest.java

```diff
@@ -14,107 +14,624 @@
+      gateway.create(dsName, 2);
+
+    } finally {
+      exln.remove();
+    }
+  }
 
-    vm4.invoke(() -> WANTestBase.doPuts(getTestMethodName() + "_RR", 1000));
+  private int createReceiver() {
+    GatewayReceiverFactory fact = cache.createGatewayReceiverFactory();
+    int port = AvailablePortHelper.getRandomAvailablePortForDUnitSite();
+    fact.setStartPort(port);
+    fact.setEndPort(port);
+    fact.setManualStart(true);
+    GatewayReceiver receiver = fact.create();
+    try {
+      receiver.start();
+    } catch (IOException e) {
+      e.printStackTrace();
+      Assert.fail(
+          "Test " + testMethodName + " failed to start GatewayReceiver on port " + port, e);
+    }
+    return port;
+  }
 
-    vm2.invoke(() -> WANTestBase.validateRegionSize(getTestMethodName() + "_RR", 1000));
+
+  private Properties getBasicMemberProperties() {
+    Properties p = new Properties();
+    p.put(ConfigurationProperties.LOG_LEVEL, "info");
+    p.put(ConfigurationProperties.MCAST_PORT, "0");
+    return p;
   }
 
-  @Test
-  public void testSenderNoSSLReceiverSSL() {
-    IgnoredException.addIgnoredException("Unexpected IOException");
-    IgnoredException.addIgnoredException("SSL Error");
-    IgnoredException.addIgnoredException("Unrecognized SSL message");
+  private void createCache(Integer locPort) {
+    Properties props = getDistributedSystemProperties();
 
-    Integer lnPort = (Integer) vm0.invoke(() -> WANTestBase.createFirstLocatorWithDSId(1));
-    Integer nyPort = (Integer) vm1.invoke(() -> WANTestBase.createFirstRemoteLocator(2, lnPort));
+    props.putAll(getBasicMemberProperties());
+    props.put(ConfigurationProperties.LOCATORS, "localhost[" + locPort + "]");
 
-    vm2.invoke(() -> WANTestBase.createReceiverWithSSL(nyPort));
+    InternalDistributedSystem ds = getSystem(props);
+    cache = CacheFactory.create(ds);
+  }
 
-    vm4.invoke(() -> WANTestBase.createCache(lnPort));
+  private void createCacheWithSSL(Integer locPort) {
+    Properties props = getDistributedSystemProperties();
+    props.putAll(getBasicMemberProperties());
+    props.putAll(getSSLProperties());
 
-    String senderId = "ln";
-    vm4.invoke(
-        () -> WANTestBase.createSender(senderId, 2, false, 100, 10, false, false, null, true));
+    props.put(ConfigurationProperties.LOCATORS, "localhost[" + locPort + "]");
 
-    String regionName = getTestMethodName() + "_RR";
-    vm2.invoke(() -> WANTestBase.createReplicatedRegion(regionName, null, isOffHeap()));
+    logger.info("Starting cache ds with following properties \n" + props);
 
-    vm4.invoke(() -> WANTestBase.startSender(senderId));
+    InternalDistributedSystem ds = getSystem(props);
+    cache = CacheFactory.create(ds);
+  }
 
-    // Verify the sender is started
-    vm4.invoke(() -> verifySenderRunningState(senderId));
+  private Properties getSSLProperties() {
+    Properties props = new Properties();
+    props.put(ConfigurationProperties.GATEWAY_SSL_ENABLED, "true");
```

Comment:
Ignore if you want: Using "setProperty" is actually more correct here. The "put" method is inherited from Map.